File: user-interface-service/src/app/adapter/ledger-service/dto/account/AccountEnrichedObject.ts
export interface IAccountEnrichedObject {
  accountId: number;
  typeId: number;
  description: string;
  accountTypeName: string;
  accountDisplayName: string;
  accountBalance: number;
  active: boolean;
  notes: string;
  creditEffect: string;
  debitEffect: string;
}


export type AccountEnrichedObject = IAccountEnrichedObject;

File: ledger-service/src/main/java/zachkingcade/dev/ledger/adapter/in/web/dto/account/AccountEnrichedObject.java
package zachkingcade.dev.ledger.adapter.in.web.dto.account;

public record AccountEnrichedObject(
        Long accountId,
        Long typeId,
        String description,
        String accountTypeName,
        String accountDisplayName,
        Long accountBalance,
        boolean active,
        String notes,
        String creditEffect,
        String debitEffect
) {
}
